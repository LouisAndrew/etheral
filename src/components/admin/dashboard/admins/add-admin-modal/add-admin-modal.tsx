import React, { useState } from 'react';
import { useForm } from 'react-hook-form';

import { Box, Heading, Flex, Button, Text } from 'rebass';
import { Label, Input } from '@rebass/forms';

type Props = {
    yes: (email: string) => void; // calls addAdmin() from admins component
    no: () => void; // close modal.
};
/**
 * basic component to make sure admin is going to be added
 */
const AddAdminModal: React.FC<Props> = ({ yes, no }) => {
    const [showSure, setShowSure] = useState(false);
    const [email, setEmail] = useState('');

    const { register, errors, handleSubmit } = useForm();

    const submit = ({ adminEmail }: any) => {
        setEmail(adminEmail);
        setShowSure(true);
    };

    return !showSure ? (
        <Box
            bg="#fff"
            p={[4]}
            width={[512]}
            sx={{ 'input[type=email]': { py: 4 } }}
            as="form"
            onSubmit={handleSubmit(submit)}
        >
            <Heading fontSize={[3]} mb={[3]}>
                Add admin member
            </Heading>
            <Label
                htmlFor="admin-email"
                sx={{ fontFamily: 'heading', fontSize: [1], mb: 2 }}
            >
                Admin email address:
            </Label>
            <Input
                type="email"
                variant="variants.authInput"
                id="admin-email"
                placeholder="New admin email address"
                name="adminEmail"
                ref={register({
                    required: true,
                })}
            />

            {errors.adminEmail && (
                <Text fontSize={1} fontFamily="body" color="misc.discount">
                    Please add a valid email address
                </Text>
            )}

            <Button width="100%" mt={5} type="submit">
                Add admin
            </Button>
            <Button width="100%" mt={2} bg="misc.discount" onClick={no}>
                Cancel
            </Button>
        </Box>
    ) : (
        <Box bg="#fff" p={[4]} width={[512]}>
            <Heading fontSize={[3]} mb={[3]}>
                Are you sure?
            </Heading>
            <Flex mt={5}>
                <Button width="100%" mr={3} onClick={() => yes(email)}>
                    Yes
                </Button>
                <Button
                    width="100%"
                    bg="misc.discount"
                    onClick={() => setShowSure(false)}
                >
                    No
                </Button>
            </Flex>
        </Box>
    );
};

export { AddAdminModal };
