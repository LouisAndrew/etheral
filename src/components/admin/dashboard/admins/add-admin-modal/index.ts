
import { AddAdminModal } from './add-admin-modal';

export default AddAdminModal;