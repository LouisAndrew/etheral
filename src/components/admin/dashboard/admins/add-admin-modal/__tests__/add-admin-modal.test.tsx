import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

// import { render, cleanup } from '@testing-library/react'
import { cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';

import { AddAdminModal } from '../add-admin-modal';

describe('AddAdminModal', () => {
    const mockyes = jest.fn((email: string) => {
        console.log(email);
    });

    const Element = <AddAdminModal yes={mockyes} no={jest.fn(() => {})} />;

    afterEach(cleanup);

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(Element, div);
    });

    /* it('renders correctly', () => {
		const { getByTestId } = render()
	}) */

    it('matches snapshot', () => {
        const run = true;

        if (run) {
            const tree = renderer.create(Element).toJSON();
            expect(tree).toMatchSnapshot();
        }
    });
});
