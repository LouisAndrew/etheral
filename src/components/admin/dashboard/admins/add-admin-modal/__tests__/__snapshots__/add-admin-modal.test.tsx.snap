// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddAdminModal matches snapshot 1`] = `
<form
  className="css-1eye3dk"
  onSubmit={[Function]}
>
  <h2
    className="css-aeusdd"
  >
    Add admin member
  </h2>
  <label
    className="css-cf12dr"
    htmlFor="admin-email"
  >
    Admin email address:
  </label>
  <input
    className="css-4y926w"
    id="admin-email"
    name="adminEmail"
    placeholder="New admin email address"
    type="email"
  />
  <button
    className="css-s17rj8"
    type="submit"
  >
    Add admin
  </button>
  <button
    className="css-14fcrsr"
    onClick={[MockFunction]}
  >
    Cancel
  </button>
</form>
`;
